entrypoint.sh

```diff
@@ -0,0 +1,221 @@
+if test -t 1; then
+    # see if it supports colors...
+    ncolors=$(tput colors)
+
+    if test -n "$ncolors"; then
+        if test $ncolors -ge 8; then
+            bold="$(tput bold)"
+            underline="$(tput smul)"
+            standout="$(tput smso)"
+            normal="$(tput sgr0)"
+            black="$(tput setaf 0)"
+            red="$(tput setaf 1)"
+            green="$(tput setaf 2)"
+            yellow="$(tput setaf 3)"
+            blue="$(tput setaf 4)"
+            magenta="$(tput setaf 5)"
+            cyan="$(tput setaf 6)"
+            white="$(tput setaf 7)"
+        fi
+        if test $ncolors -ge 9; then
+            gray="$(tput setaf 8)"
+        fi
+    fi
+fi
+# end color support
+
+sub_help() {
+    echo "Usage: $PROG_NAME <subcommand> [options]"
+    echo "Subcommands:"
+    echo "    figures   Generate figures from the paper; usage: $PROG_NAME figures [--rerun-bootstrap] <figure_name|clean>"
+    echo "               --rerun-bootstrap: recomputes bootstrap estimates from scratch (expensive)"
+    echo "               figure_name: recreates a specific figure, or if 'all' is specified all of them."
+    echo "                 available options: figure1, figure2, figure3, figure4, supplemental1, all"
+    echo "               clean: removes all intermediate and final figure artifacts, and the bootstrap data cache"
+    echo ""
+    echo "    shell     Creates a (bash) shell within the environment"
+    echo ""
+}
+
+
+# ====================================================================================
+# === 'shell' subcommand
+# ====================================================================================
+
+sub_shell() {
+    bash
+}
+
+
+# ====================================================================================
+# === 'figures' subcommand
+# ====================================================================================
+
+build_figure1() {
+    echo "*** Building figure 1..."
+    Rscript -e "library(rmarkdown); rmarkdown::render('figure1.Rmd')"
+    return $?
+}
+build_figure2() {
+    if [[ $1 -eq 1 ]]; then
+        rm -f /app/figure_notebooks/tmp_files/fig2_tmp/fig2.RData
+    fi
+    echo "*** Building figure 2..."
+    Rscript -e "library(rmarkdown); rmarkdown::render('figure2.Rmd')"
+    return $?
+}
+build_figure3() {
+    if [[ $1 -eq 1 ]]; then
+        rm -f /app/figure_notebooks/tmp_files/fig3_tmp/all_bootstrap_df.tsv
+    fi
+    echo "*** Building figure 3..."
+    Rscript -e "library(rmarkdown); rmarkdown::render('figure3.Rmd')"
+    return $?
+}
+build_figure4() {
+    if [[ $1 -eq 1 ]]; then
+        rm -f /app/figure_notebooks/tmp_files/fig4_tmp/*.tsv
+    fi
+    echo "*** Building figure 4..."
+    Rscript -e "library(rmarkdown); rmarkdown::render('figure4.Rmd')"
+    return $?
+}
+build_supplemental1() {
+    echo "*** Building supplemental 1..."
+    Rscript -e "library(rmarkdown); rmarkdown::render('supp_fig1.Rmd')"
+    return $?
+}
```

Comment:
Very minor. Looks like it's the same code but changes the notebook path and output file. Maybe in the future you can have a single function, and call it differently from the `case...in` below. This is super minor and even stylistic and you can omit it.

Reply:
I agree, I think for now I will keep it like this in case I need to add things. If it seems stable, then I will refactor this to be a bit cleaner.